src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -28,16 +32,30 @@ public HttpHandlerResourceV3Provider()
 
             HttpHandlerResourceV3 curResource = null;
 
-            var clientHandler = CreateCredentialHandler(source.PackageSource);
+            if (source.PackageSource.IsHttp)
```

Comment:
No, the contract for resource providers is that they will not return a resource if it does not apply to the source or the source does not support it.

In this case a local source does not support http, so there is no http resource.